plugins/prefixfree.viewport-units.js

```diff
@@ -24,19 +24,39 @@ if(!units.length) {
 
 StyleFix.register(function(css) {
 	var w = innerWidth, h = innerHeight;
-	
-	return css.replace(RegExp('\\b(\\d+\\.?\\d*)(' + units.join('|') + ')\\b', 'gi'), function($0, num, unit) {
+	// For help with this monster regex: see debuggex.com: https://www.debuggex.com/r/cpzpAHiWPagP3Zru
+	return css.replace(RegExp('(-?[a-z]+(?:-[a-z]+)*\\s*:\\s*)\\b([0-9]*\\.?[0-9]+)(' + units.join('|') + ')\\b(?:\\s*;\\s*\\1\\b(?:[0-9]*\\.?[0-9]+)(?:px)\\b)?;?', 'gi'), function($0, property, num, unit) {
```

Comment:
There's a small hitch here, in that the regexp doesn't handle comments. You'd need something along the lines of

```JS
RegExp(
    "(-?[a-z]+(?:-[a-z]+)*)" +
    "\\s*(?:\\/\\*(?:(?!\\*\\/).)*?\\*\\/\\s*)*:" +
    "\\s*(?:\\/\\*(?:(?!\\*\\/).)*?\\*\\/\\s*)*([0-9]*\\.?[0-9]+)(" +
        units.join('|') +
    ")\\b(?:" +
        "\\s*(?:\\/\\*(?:(?!\\*\\/).)*?\\*\\/\\s*)*;" +
        "\\s*(?:\\/\\*(?:(?!\\*\\/).)*?\\*\\/\\s*)*\\1" +
        "\\s*(?:\\/\\*(?:(?!\\*\\/).)*?\\*\\/\\s*)*:" +
        "\\s*(?:\\/\\*(?:(?!\\*\\/).)*?\\*\\/\\s*)*[0-9]*\\.?[0-9]+px\\b" +
    ")?" +
    "\\s*(?:\\/\\*(?:(?!\\*\\/).)*?\\*\\/\\s*)*;?",
    'gi'
)
```

I've split the source in multiple lines for readability, but for optimal gzippability, you'd want both pre- and post- `unit` parts in one chunk each.

I've also corrected what I belived was mistakes (extra `\b` and a missing space).

The regexp was generated [with this gist](https://gist.github.com/pygy/3b50cc0e41088a80b5af), using [`compose-regexp`](https://github.com/pygy/compose-regexp.js) which makes it easy to read, write and maintain complex regexps.

```JS
__ = sequence( // space and comments
    /\s*/,
    greedy('*',
        '/*',
        frugal('*',
            avoid('*/'),
            /./
        ),
        '*/',
        /\s*/
    )
)

matcher = sequence(
    capture(/-?[a-z]+(?:-[a-z]+)*/),
     __, ':', __,
    capture(/[0-9]*\.?[0-9]+/),
    capture('UNITS'),
    /\b/,
    maybe(
        __, ';', __, ref(1), __, ':', __,
        /[0-9]*\.?[0-9]+px\b/
    ),
    __, /;?/
)
```

Interested in a PR with the updated RegExp? It is much larger but should gzip very well.